### PR TITLE
projects: fmcdaq2: initial version

### PR DESCRIPTION
**Pull request by amiclaus**
Update FMCDAQ2 project to use latest platform drivers and AXI core implementations.
Folder structure is kept similar to the updated projects.

Add makefile support for FMCDAQ2 project (Makefile, src.mk).

NOTE: Current implementation supports the following carrier boards: KCU105, VC707, KC705

Signed-off-by: Antoniu Miclaus antoniu.miclaus@analog.com

### CODE REVIEW

File: projects/fmcdaq2/src/app/fmcdaq2.c [hunk 1]
```diff
@@ -206,77 +204,142 @@ int fmcdaq2_reconfig(struct ad9144_init_param *p_ad9144_param,
...
+		.mode = SPI_MODE_0
+	};
 
-#ifdef ALTERA
-	ad9523_spi_param.type = NIOS_II_SPI;
-	ad9144_spi_param.type = NIOS_II_SPI;
-	ad9680_spi_param.type = NIOS_II_SPI;
+#ifndef ALTERA_PLATFORM
+	struct xil_spi_init_param xil_spi_param = {
+#ifdef PLATFORM_MB
```
Comment (adrimbarean) [inline projects/fmcdaq2/src/app/fmcdaq2.c]: Where is `PLATFORM_MB` defined here? It would usually be in the config.h file. Is it a BSP define?

Comment (amiclaus) [reply projects/fmcdaq2/src/app/fmcdaq2.c]: BSP define :)

File: projects/fmcdaq2/Makefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+TARGET := fmcdaq2
+ifeq ($(OS), Windows_NT)
+include ../../tools/scripts/windows.mk
+else
+include ../../tools/scripts/linux.mk
+endif
```
Comment (adrimbarean) [inline projects/fmcdaq2/Makefile]: nitpick: no new line at the end of file

File: projects/fmcdaq2/src.mk [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	$(DRIVERS)/axi_core/jesd204/axi_jesd204_tx.c			\
+	$(DRIVERS)/axi_core/jesd204/xilinx_transceiver.c		\
+	$(DRIVERS)/frequency/ad9523/ad9523.c				\
+	$(DRIVERS)/adc/ad9680/ad9680.c					\
+	$(DRIVERS)/dac/ad9144/ad9144.c					\
+	$(NO-OS)/util/util.c
+SRCS +=	$(PLATFORM_DRIVERS)/axi_io.c					\
+	$(PLATFORM_DRIVERS)/spi.c					\
+	$(PLATFORM_DRIVERS)/gpio.c					\
+	$(PLATFORM_DRIVERS)/delay.c
```
Comment (adrimbarean) [inline projects/fmcdaq2/src.mk]: Is it good practice to have all the `SRCS=` or can we condense them all into one?

Comment (amiclaus) [reply projects/fmcdaq2/src.mk]: The pattern is kept similar to the other projects' makefiles.
The `srcs` is somehow splitted based on the modules that are included (in this case, based on the env vars `DRIVERS` for device drivers and `PLATFORM_DRIVERS` for platform specific drivers).

**Comment by amiclaus** [comment]
V2:
- add new line at the end of file
- remove `/fmcdaq2` folder from no-OS root folder when performing the project move in the first commit

**Comment by amiclaus** [comment]
V3:
- add platform specific gpio offset in `parameters.h` header file

File: projects/fmcdaq2/src/devices/adi_hal/parameters.h [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+#include "app_config.h"
+#ifdef ALTERA_PLATFORM
+#include "system.h"
+#else
+#include "xparameters.h"
+#endif
+
+#ifndef ALTERA_PLATFORM
+#ifdef PLATFORM_MB
```
Comment (dbogdan) [inline projects/fmcdaq2/src/devices/adi_hal/parameters.h]: Where are PLATFORM_MB and PLATFORM_ZYNQMP defined?

Comment (amiclaus) [reply projects/fmcdaq2/src/devices/adi_hal/parameters.h]: There is a similar question above.
These are BSP defines, you can find them in `xparameters.h`.

**Comment by amiclaus** [comment]
V4:

- fix code style issues

File: projects/fmcdaq2/src/devices/adi_hal/parameters.h (hunk 4 above)
Comment (dbogdan) [reply projects/fmcdaq2/src/devices/adi_hal/parameters.h]: OK, sorry.

**Comment by amiclaus** [comment]
V5:
- fix conversion from KHz to Hz for altera platform specific parameters
- use only `gpio_direction_output` function to set gpio values due to lack of implementation for `set_value` function in the altera platform gpio driver
- make sure that lane rate parameter is initialized for the ADC
- map missing registers required for `adxcvr` altera platform specific structures

**Comment by dbogdan** [comment]
Let's wait until Altera support is finalized before merging to master.

**Comment by amiclaus** [comment]
V6:
-  fix TX_CORE_BASEADDR in `parameters.h` (add 0x4000 offset)

**Pull request merged**